llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -2742,7 +2742,11 @@ static Instruction *foldSelectWithFCmpToFabs(SelectInst &SI,
     // Note: We require "nnan" for this fold because fcmp ignores the signbit
     //       of NAN, but IEEE-754 specifies the signbit of NAN values with
     //       fneg/fabs operations.
-    if (!SI.hasNoSignedZeros() || !SI.hasNoNaNs())
+    if (!SI.hasNoNaNs())
+      return nullptr;
+
+    bool functionHasNoSignedZeroes = SI.getParent()->getParent()->hasFnAttribute("no-signed-zeros-fp-math");
```

Comment:
I think `isFast()` implies `noSignedZeros`, can you just use that?

Reply:
It's better to look for the individual semantic modes. While `isFast()` implies `noSignedZeroes`, it is possible to enable `noSignedZeroes` without enabling other fast-math flags, so `isFast()` might return false even when `noSignedZeros` is enabled.